service/k8sclient/kubectl/kubectl.go

```diff
@@ -32,30 +34,116 @@ import (
 	"github.com/percona-platform/dbaas-controller/utils/logger"
 )
 
+const (
+	// FIXME: change to a real default kubectl.
+	defaultKubectl = "minikube kubectl --"
+)
+
 // KubeCtl wraps kubectl CLI with version selection and kubeconfig handling.
 type KubeCtl struct {
 	l   logger.Logger
 	cmd []string
 }
 
 // NewKubeCtl creates a new KubeCtl object with a given logger.
-func NewKubeCtl(l logger.Logger) *KubeCtl {
-	// TODO Handle kubectl versions https://jira.percona.com/browse/PMM-6348
+func NewKubeCtl(ctx context.Context) (*KubeCtl, error) {
+	l := logger.Get(ctx)
+	l = l.WithField("component", "kubectl")
 
-	// TODO Handle kubeconfig https://jira.percona.com/browse/PMM-6347
-
-	cmd := []string{"dbaas-kubectl-1.16"}
-	path, err := exec.LookPath(cmd[0])
-	l.Debugf("path: %s, err: %v", path, err)
-	if e, ok := err.(*exec.Error); err != nil && ok && e.Err == exec.ErrNotFound {
-		cmd = []string{"minikube", "kubectl", "--"}
+	// Handle kubectl versions
+	cmd, err := getKubectlCmd(ctx)
+	if err != nil {
+		l.Debugf("Cannot find kubectl binary: %s", err)
```

Comment:
In general, we should either handler errors (by logging them, for example), or return them, but not both. That error will be logged by an interceptor, so let's just return it.